webpack/assets/javascripts/react_app/common/helpers.js

```diff
@@ -3,6 +3,20 @@ import { snakeCase, camelCase } from 'lodash';
 import URI from 'urijs';
 import { translate as __ } from './I18n';
 
+/**
+ * Our API returns non-ISO8601 dates.
+ * This method converts those strings into ISO8601 format
+ * @param {String} date - non-ISO date to convert
+ * @returns {String} ISO8601 compatible date string
+ */
+export const isoCompatibleDate = date => {
+  if (typeof date === 'string') {
```

Comment:
Would it make sense to add a check here that the date is in the format you expect it to be? Otherwise there may be other cases we return date in some third format that this will mess up.

Reply:
It feels to me unsafe to manipulate string dates based on manual replaces.
The Date constructor should be able to create a Date instance from multiple types of string formats.
I'm wondering if such an approach would work, can you try something like this?:
```js
const myDate = new Date(dateString);

// or maybe we don't even need this function
context.intl.formatRelative(new Date(date))
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

Reply:
That doesn't really help. In the Firefox console:
```
new Date("2019-03-14 09:26:17 -0400")
Invalid Date
```

That's the root cause of the issue: `new Date()` can't parse this. I suspect the library internally does the conversion as well. The docs you linked also have a warning about this exact case:
> Note: parsing of date strings with the Date constructor (and Date.parse, they are equivalent) is strongly discouraged due to browser differences and inconsistencies. Support for RFC 2822 format strings is by convention only. Support for ISO 8601 formats differs in that date-only strings (e.g. "1970-01-01") are treated as UTC, not local.

Reply:
@tbrisker can you think of an example? I think this method should take dates in a "reasonable format" ie passing a string "2019-03-14" won't break anything. Giving the component a bogus date in general would cause it to not work in any case.

@sharvit as Ewoud mentioned this is the problem I am working around with this change.

Reply:
I would prefer just to make sure we get the format we expect before manipulating it, there may be other formats we are not thinking about right now.

Reply:
@tbrisker updated